include/SFML/Graphics/Rect.hpp

```diff
@@ -44,94 +44,152 @@ class Rect
 {
 public:
     ////////////////////////////////////////////////////////////
-    /// \brief Default constructor
+    /// \brief Construct the rectangle from position and size.
     ///
-    /// Creates an empty rectangle (it is equivalent to calling
-    /// Rect(0, 0, 0, 0)).
+    /// \param position Position of the rectangle
+    /// \param size     Size of the rectangle
     ///
     ////////////////////////////////////////////////////////////
-    constexpr Rect();
+    constexpr Rect(Vector2<T> rectanglePosition = Vector2<T>(), Vector2<T> rectangleSize = Vector2<T>());
 
     ////////////////////////////////////////////////////////////
-    /// \brief Construct the rectangle from position and size
+    /// \brief Construct the rectangle from another type of rectangle.
     ///
-    /// Be careful, the last parameter is the size,
-    /// not the bottom-right corner!
+    /// This constructor doesn't replace the copy constructor
     ///
-    /// \param position Position of the top-left corner of the rectangle
-    /// \param size     Size of the rectangle
+    /// \param rectangle Rectangle to convert
     ///
     ////////////////////////////////////////////////////////////
-    constexpr Rect(const Vector2<T>& position, const Vector2<T>& size);
+    template <typename U>
+    constexpr explicit Rect(Rect<U> rectangle);
 
     ////////////////////////////////////////////////////////////
-    /// \brief Construct the rectangle from another type of rectangle
+    /// \brief Get the left edge of the rectangle
     ///
-    /// This constructor doesn't replace the copy constructor,
-    /// it's called only when U != T.
-    /// A call to this constructor will fail to compile if U
-    /// is not convertible to T.
+    /// \return Coordinate of the left edge of the rectangle
     ///
-    /// \param rectangle Rectangle to convert
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr T getLeft() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Get the top edge of the rectangle
+    ///
+    /// \return Coordinate of the top edge of the rectangle
     ///
     ////////////////////////////////////////////////////////////
-    template <typename U>
-    constexpr explicit Rect(const Rect<U>& rectangle);
+    [[nodiscard]] constexpr T getTop() const;
```

Comment:
We try to use `[[nodiscard]]` rather conservatively, see discussion in https://github.com/SFML/SFML/pull/1883.

Reply:
So do I remove it?

Reply:
Yes (same for other similar places).